packages/babel-parser/src/parser/expression.js

```diff
@@ -2269,4 +2295,36 @@ export default class ExpressionParser extends LValParser {
       this.state.topicContext.maxTopicIndex >= 0
     );
   }
+
+  makePipelineHead(left: N.Expression): N.Expression {
+    const node = this.startNode();
+    node.head = left;
+
+    return this.finishNode(node, "PipelineHead");
+  }
+
+  parseFSharpPipelineBody(
+    op: TokenType,
+    prec: number,
+    noIn: ?boolean,
+  ): N.Expression {
+    const startPos = this.state.start;
+    const startLoc = this.state.startLoc;
+
+    const node = this.startNode(startPos, startLoc);
+
+    if (this.lookahead().type === tt.arrow) {
+      this.state.potentialArrowAt = this.state.start;
+    }
```

Comment:
This in particular I'm not sure is the best way to go about it.

Reply:
Actually, the conditional wasn't needed at all. Always setting a potential arrow makes it parse correctly.